Feature/sref

BinbinZhou-NOAA opened the pull request:
## Pull Request Testing ##

- [y ] Describe testing already performed for this Pull Request:</br>

- [ y] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ y] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ y] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [y ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ y] Review the source issue metadata (required labels, projects, and milestone).
- [ y] Complete the PR description above.
- [ y] Ensure the PR title matches the feature branch name.
- [y ] Check the following:
- [ y]  Instructions provided on how to run
- [ y]  Developer's name is replaced by ${user} where necessary throughout the code
- [y ]  Check that the ecf file has all the proper definitions of variables
- [y ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ y]  Check to see that the output directory structure is followed
- [y ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

BinbinZhou-NOAA committed:
Sent first version of sref package

BinbinZhou-NOAA committed:
new update for SREF

BinbinZhou-NOAA committed:
New update

BinbinZhou-NOAA committed:
Add test ecf directory

BinbinZhou-NOAA committed:
Update instruction file

BinbinZhou-NOAA committed:
Add new CTC verification for CNV

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref

BinbinZhou-NOAA committed:
update PointStat_fcstSREF_obsPREPBUFR_cnv.conf

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref

ShelleyMelchior-NOAA commented on jobs/mesoscale/stats/JEVS_MESOSCALE_STATS:
```diff
@@ -1 +1,79 @@
+
+export HOMEevs=${HOMEevs:-${PACKAGEROOT}/evs.${evs_ver}}
+export EXECevs=${EXECevs:-$HOMEevs/exec}
+export FIXvevs=${FIXevs:-$HOMEevs/fix}
```
@BinbinZhou-NOAA ... I see a typo in line 45 ... FIXvevs=

BinbinZhou-NOAA committed:
fix a typo in J-job

BinbinZhou-NOAA committed:
Merge branch 'feature/sref' of https://github.com/BinbinZhou-NOAA/EVS into feature/sref

* 'feature/sref' of https://github.com/BinbinZhou-NOAA/EVS: (335 commits)
  Change precip spatial map iamge name
  Change METPLUS_CONF name
  Get GHRSST Median data from dcom
  Add sea-ice concentration RMSE and Bias plotting
  Add plotting jobs
  Update neighborhood sizes to plot
  Update plotting script for g2o sfc CAPE jobs
  Add support for plotting FBAR & OBAR
  Remove setting of fcst/obs_var_thresh
  Fix descriptions
  Add threshold average plot
  Append plot type to image name
  Add missing )
  Add valid hour average plot
  Update pull_request_template.md
  Use ADPUPA for plotting
  Change precip spatial maps plot specs
  Convert spatial precip maps to gif
  Add regions for precip spatial maps
  Adjust metfra walltime
  Adjust walltimes
  Use neighborhood environment variable
  Adjust neighborhood groups to balance run times
  Add check for UKMET data
  Rename job groups and add "assemble_data"
  Adjustments for averaging jobs
  remove evs.v1.0 sub-directory
  Adjust walltimes
  Update stats job script creation scripts
  Change log and conf files names
  Adjust init time for other forecast hours
  Remove pycache
  Add Ozone job not running for certain international models
  Fix typo in logging message
  Add exit statement
  Unify METplus out file naming
  Remove BUCKET_INTERVAL config setting
  Add check for .stat files for 2m TMP Anom jobs
  Add reworked gather jobs
  Rename pb2nc output file
  Fix bug in grid2obs daily average script
  Add reworked generate jobs
  Rework how job scripts are run and created
  Add missing loop increment
  Add updates related to reformat jobs
  Add checking for model files for reformat jobs
  Fix typo
  Rework truth file checking
  Add checking for reformat obs files
  Remove getting OSI-SAF weekly file
  Add checking obs and fcst file existing
  Use CCPA lons and lats for plotting
  Change large stat file name to lower case
  Add preformance diagram
  Add precipitation spatial maps
  Update ecf files
  Update NARRE
  Censor forecast VIS value gt16090
  Remove sea ice weekly average stats
  Fix Visibility and Ceiling thresholds
  Add plots types for plotting jobs
  Switch CMC and FNMOC colors
  Make plot adjustments
  Add lead_by_level plot
  Removed unnecessary plot title setting
  Add addtional debug logging statement
  Add function for getting list of levels in vertical profile
  Fix daily avg SST plot title info
  Write out RAOBS and aircraft stats separately
  Change sea ice concentration job name
  Adjust walltimes
  Minor adjustments for daily avg jobs
  Close and save netcdf files
  Use valid time in two digit format
  Add SST obs name for plotting job
  Rework copying to COMOUT
  Update Total Ozone title variable
  Adjust walltimes again
  Do another rework of copying to COMOUT
  Update COMOUT directories
  Adjust walltime
  Change SST plotting init hour
  Initial commit of plots ecf scripts and configs
  Change names of plot tar files
  Set KEEPDATA to NO for plotting ecf scripts
  Rename plotting ecf scripts and configs
  Initial commit of stat_by_level plot
  Switch to using ERA5 climo
  Change way variable units are retrieved
  Fix wrong function calls
  Fix logging statements
  Add GET-D ALEXI data
  Rework copying to COMOUT
  Rework set up for copmuting anomalies and averaged files
  Use COMIN files for weekly average files
  Adjust aggeregation method calculation
  Add aggegration for computing average
  Add in plotting of data for lead_by_date
  Change class reference
  Make adjustment to lead_by_date plot specs
  Add confidence intervals for lead_average
  Change number of columns in time series legend
  Add 'days' in time series legend label
  Change line plot settings
  Add average and difference lines to lead average plot
  Update pull_request_template.md
  Update pull_request_template.md
  Update pull_request_template.md
  Use values from stats dataframe to plot time series
  Set stat dataframe to event equalized values
  Make lead average plot adjustments
  Make time series plot adjustments
  Make adjustments for lead by date plot
  Put error messages in all caps
  Add check for number of models being plotted
  Initial commit of lead by date plot
  Fix function description
  Change build_df inputs
  Initial commit of lead average plot
  Fix typo
  Add sst verification to production configs
  Make some updates for averaging files
  Add grid2grid sst plots jobs
  Save fcst var for sst as SST_DAILYAVG
  Add grid2grid sst stats jobs
  Add ghrsst_median global_det atmos prep work
  Add sea_ice concentration plotting jobs
  Add daily and weekly sea-ice concentration stats jobs
  Add _DAILYAVG to daily averaged variables
  Update sea_ice configuration settings
  Add jobs for creating daily and weekly sea-ice averages
  Update sea_ice settings in standalone config
  Verify 00Z cycle for sea_ice for GFS
  Add getting of model files for sea_ice averaging
  Add getting of OSI-SAF weekly avg files
  Rework osi_saf prep
  Add creation of weekly avg osi_saf files
  Add collection of osi_saf data for verification
  Add wind shear verification
  Change ecf scripts to support only dev testing
  Update ecf and jobs scripts for ops
  Create obs type AIRUPA (RAOBS + aircraft)
  new update for NARRE
  Change plots tar file name
  Fix typo in GFS analysis path for CFS g2g
  Set KEEPDATA to NO
  Add creation of images tar file and copying to COMOUT
  Make adjustments for grid2obs plots
  Adjust Mixed layer CAPE plotting jobs
  Add thresholds for CAPE RMSE and Bias plotting jobs
  Use INIT_SEQ in grid-to-obs stats METplus confs
  Add CFS to global_det atmos prep
  Change dev testing DATAROOT and COMIN/OUT locations
  Update path to masking regions files
  Rework getting variable name for plots
  Fix g2g snow plotting jobs variable name and level
  Add new statistic names
  Fix cfs model data paths
  Fix bug
  Add POD and CSI calculations
  Add SRATIO calculation
  Add grid information for grid-to-obs plotting
  Fix incorrectly set use case for grid-to-obs plots
  Include threshold in savefig name
  Fix typo for WEASD variable name
  Update determining of units for plot
  Fix typo
  Update plot title for unrecongized level
  Minor update to parsed file name
  Update adjusting of plot y-axis
  Fix bug in determining units
  Fix bug
  Use certain COMIN paths for stats only
  Adjust production stat file path for plotting
  Adjust global_det atmos plots COMIN/COMOUT
  Add COMIN observation paths
  Change envir to dev; add special dev COMIN/COMOUT
  Update global_det atmos COMINcmc path
  Update global_det atmos stats COMIN and COMOUT
  Initial commit for fixing COMIN and COMOUT paths
  Add loading to MET and METplus for plotting
  Make some adjustments to support plotting ecf jobs
  Change file permissions
  Remove exit statements
  Switch start and end date plotting range
  Add initial commit of plotting ecf files and configs
  Add ecf files and configs for international models
  Fix setting of global_det atmos stats COMOUT sub directories
  Set KEEPDATA to NO for global_det atmos prep
  Source config for global_det atmos grid2obs stats
  Add gfs grid-to-obs ecf file and config
  Make GFS grid-to-grid atmos ecf job and config
  Slight rework of global_det atmos prep
  Change test dev directory path
  Add NET variable for global_det atmos prep
  Fix typos
  Remove NaN values for getting forecast var units
  Add exist_ok for directory creation
  Add grid-to-obs plotting jobs
  Set levels to all caps
  ...

BinbinZhou-NOAA commented:
The typoed is not used so can be removed. I have deleted that line
and updated the job file in the folk

Thanks!

Binbin

On Tue, Nov 1, 2022 at 11:04 AM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In jobs/mesoscale/stats/JEVS_MESOSCALE_STATS
> <https://github.com/NOAA-EMC/EVS/pull/29#discussion_r1010539491>:
>
> > +export COMINprepbufr=${COMINprepbufr:-$(compath.py  obsproc/${obsproc_ver})}
> +export COMINccpa=${COMINccpa:-$(compath.py  ccpa/${ccpa_ver})}
> +export COMINsrefmean=${COMINsrefmean:-$(compath.py  $NET/$evs_ver)}/prep/$COMPONENT
> +
> +export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})}/$STEP/$COMPONENT
> +mkdir -p $COMOUT
> +
> +
> +export COMOUTfinal=${COMOUT}/${MODELNAME}.${VDATE}
> +export COMOUTsmall=${COMOUT}/${RUN}.${VDATE}/${MODELNAME}/${VERIF_CASE}
> +mkdir -p $COMOUTfinal $COMOUTsmall
> +
> +
> +export HOMEevs=${HOMEevs:-${PACKAGEROOT}/evs.${evs_ver}}
> +export EXECevs=${EXECevs:-$HOMEevs/exec}
> +export FIXvevs=${FIXevs:-$HOMEevs/fix}
>
> I see a typo in FIXevs=
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/29#pullrequestreview-1163501520>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFJT4F6ZZRFVNGD6VADWGEWPHANCNFSM6AAAAAARONK4RE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Thanks Binbin. I think you also accidentally introduced some *.o * files to the `ecf/mesocale.dev/stats` directory.

BinbinZhou-NOAA committed:
Delete useless files in ecf.dev/stats

BinbinZhou-NOAA commented:
Shelley,

   Yes, in ecf/mesoscale.dev/stats, there are some files that are output of
testing.
All of those files are useless and have been deleted. The feature/sref has
been updated.

Thanks!

Binbin

On Tue, Nov 1, 2022 at 12:38 PM Shelley Melchior ***@***.***>
wrote:

> Thanks Binbin. I think you also accidentally introduced some *. files to
> the ecf/mesocale.dev/stats directory.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/29#issuecomment-1298809871>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFICNWJKGLOSUFQQIADWGFBRBANCNFSM6AAAAAARONK4RE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
Update: remove ens production from EnsampleStat and add GenEnsProd

PerryShafran-NOAA commented:
Hi, @BinbinZhou-NOAA -

I see you made some important updates to add GenEnsProd to the SREF verification.  Are you now ready for me to begin testing, or is there still more you need to add to this PR?

Thanks!

Perry

BinbinZhou-NOAA commented:
Perry,

     I just planned to send a notification email to you.
   Yes, I just made some critical updates for SREF's METplus conf files to
complies future METplus (v11.0.0):
    (1) Removed ensemble production from the EnsembleStat
    (2) Added new conf files for the new tool GenEnsProd in which all of
the ensemble products are generated
    (3) Changed the big stat files by adding ens.stat. at the head of
original big stat files
   Since this is a big updates, please re-tested the SREF package again
although the new updated package
   is running fine on my account.

    In addition,  the NARRE also was updated by adding the evs.stats. at
the head of big stat files.
This is a minor update, so it is no need to test this again.

Thanks!

Binbin


On Wed, Nov 16, 2022 at 10:12 AM PerryShafran-NOAA ***@***.***>
wrote:

> Hi, @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -
>
> I see you made some important updates to add GenEnsProd to the SREF
> verification. Are you now ready for me to begin testing, or is there still
> more you need to add to this PR?
>
> Thanks!
>
> Perry
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/29#issuecomment-1317175177>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFMEWKQF4NYURZVX3SDWIT2UBANCNFSM6AAAAAARONK4RE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

PerryShafran-NOAA commented:
Hi, Binbin,

Can you provide the instructions to run this code?  I know you provided them previously, but not in this PR conversation.  Please provide the instructions here.

Thanks!

Perry

BinbinZhou-NOAA committed:
new update on 20221116

BinbinZhou-NOAA commented:
Perry,

  The instruction to run SREF is
https://github.com/BinbinZhou-NOAA/EVS/blob/feature/sref/ecf/mesoscale.dev/stats/instruction_test.txt
Simply say, go to  /lfs/h2/emc/vpppg/noscrub/$USER/EVS/ecf/
mesoscale.dev/stats, submit following 3 ecf jobs:
qsub jevs_sref_cnv_stats.ecf
qsub jevs_sref_grid2obs_stats.ecf
qsub jevs_sref_precip_stats.ecf

  I just updated the codes in SREF github folk. Should I request a new pull
before you test?

Binbin


On Wed, Nov 16, 2022 at 1:52 PM PerryShafran-NOAA ***@***.***>
wrote:

> Hi, Binbin,
>
> Can you provide the instructions to run this code? I know you provided
> them previously, but not in this PR conversation. Please provide the
> instructions here.
>
> Thanks!
>
> Perry
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/29#issuecomment-1317512999>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFJ6BHBSQFNG6GQMS4DWIUUN3ANCNFSM6AAAAAARONK4RE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

PerryShafran-NOAA commented:
Starting to check your PR now.

Perry

PerryShafran-NOAA commented:
Hi, Binbin,

I ran your codes yesterday.  Please validate the results by looking at the files in the following directories, and letting me know if the results of these stat files compare with your own run:

/lfs/h2/emc/vpppg/noscrub/perry.shafran/com/evs/v1.0/stats/mesoscale/atmos.20221119/sref/grid2obs
/lfs/h2/emc/vpppg/noscrub/perry.shafran/com/evs/v1.0/stats/mesoscale/atmos.20221119/sref/precip
/lfs/h2/emc/vpppg/noscrub/perry.shafran/com/evs/v1.0/stats/mesoscale/sref.20221119

Thanks!

Perry

BinbinZhou-NOAA commented:
Perry,

  After carefully reviewed the files you generaged and compared with the
same files generated in
my routinely run, your files are in good shape.

Here are my file directory
/lfs/h2/emc/vpppg/noscrub/binbin.zhou/com/evs/v1.0/stats/mesoscale/atmos.20221119/sref/grid2obs
/lfs/h2/emc/vpppg/noscrub/binbin.zhou/com/evs/v1.0/stats/mesoscale/atmos.20221119/sref/precip
/lfs/h2/emc/vpppg/noscrub/binbin.zhou/com/evs/v1.0/stats/mesoscale/sref.20221119

The files in my directories are exactly same as yours.

Thanks!

Binbin

On Tue, Nov 22, 2022 at 7:11 AM PerryShafran-NOAA ***@***.***>
wrote:

> Hi, Binbin,
>
> I ran your codes yesterday. Please validate the results by looking at the
> files in the following directories, and letting me know if the results of
> these stat files compare with your own run:
>
>
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/com/evs/v1.0/stats/mesoscale/atmos.20221119/sref/grid2obs
>
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/com/evs/v1.0/stats/mesoscale/atmos.20221119/sref/precip
>
> /lfs/h2/emc/vpppg/noscrub/perry.shafran/com/evs/v1.0/stats/mesoscale/sref.20221119
>
> Thanks!
>
> Perry
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/29#issuecomment-1323568344>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFPAQLZ7VOUGTITPRCTWJSZ6NANCNFSM6AAAAAARONK4RE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

PerryShafran-NOAA commented:
Excellent!  I think that we can approve this PR and merge this one in.  

Thanks!

Perry
